src/main/java/com/minerarcana/floralchemy/block/BlockCrystalthorn.java

```diff
@@ -0,0 +1,174 @@
+	private Tuple<ResourceLocation, Integer> crystal;
+	private ItemBlock itemBlock;
+	private ItemStack cachedCrystalStack = ItemStack.EMPTY;
+
+	public BlockCrystalthorn(Tuple<ResourceLocation, Integer> entry) {
+		super(Material.PLANTS);
+		this.setDefaultState(this.blockState.getBaseState().withProperty(AGE, 0).withProperty(BERRIES, false));
+		this.setTranslationKey("crystalthorn");
+		this.setTickRandomly(true);
+		this.crystal = entry;
+	}
+
+	@SideOnly(Side.CLIENT)
+	public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
+		super.addInformation(stack, worldIn, tooltip, flagIn);
+		tooltip.add(this.getCrystalStack().getDisplayName());
+	}
+
+	@Override
+	public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn,
+			EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+		if (state.getValue(AGE) == maxAge && state.getValue(BERRIES)) {
+			ItemStack stack = this.getCrystalStack();
+			if (!stack.isEmpty()) {
+				if (playerIn.addItemStackToInventory(stack)) {
+					playerIn.attackEntityFrom(DamageSource.CACTUS, 3F);
+					worldIn.setBlockState(pos, state.withProperty(BERRIES, false));
+				}
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public void randomTick(World worldIn, BlockPos pos, IBlockState state, Random random) {
+		if (random.nextBoolean()) {
+			if (state.getValue(AGE) < maxAge) {
+				worldIn.setBlockState(pos, state.withProperty(AGE, state.getValue(AGE) + 1));
+				return;
+			}
+			if (!state.getValue(BERRIES) && random.nextInt(10) == 0) {
+				worldIn.setBlockState(pos, state.withProperty(BERRIES, true));
+			}
+		}
+	}
+
+	@Override
+	public IBlockState getStateFromMeta(int meta) {
+		if (meta == maxAge + 1) {
+			return this.getDefaultState().withProperty(BERRIES, true).withProperty(AGE, maxAge);
+		}
+		return this.getDefaultState().withProperty(AGE, meta);
+	}
+
+	@Override
+	public int getMetaFromState(IBlockState state) {
+		return state.getValue(BERRIES) ? state.getValue(AGE) : state.getValue(AGE) + 1;
+	}
+
+	@Override
+	public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos) {
+		return AABB[state.getValue(AGE)];
+	}
+
+	@Override
+	protected BlockStateContainer createBlockState() {
+		return new BlockStateContainer(this, new IProperty[] { AGE, BERRIES });
+	}
+
+	@Override
+	public void harvestBlock(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state,
+			@Nullable TileEntity te, ItemStack stack) {
+		player.attackEntityFrom(DamageSource.CACTUS, 3F);
+		super.harvestBlock(worldIn, player, pos, state, te, stack);
+	}
+	
+	@Override
+	public void onEntityCollision(World worldIn, BlockPos pos, IBlockState state, Entity entityIn) {
+		entityIn.attackEntityFrom(DamageSource.CACTUS, 1F);
+	}
+
+	@SideOnly(Side.CLIENT)
+	@Override
+	public int colorMultiplier(IBlockState state, IBlockAccess world, BlockPos pos, int tintIndex) {
```

Comment:
This method needs cachin' as I don't think tint changes often, but I'm hella sure the ColourHelper isn't changing.